include/boost/beast/http/impl/read.ipp

```diff
@@ -34,17 +35,30 @@ namespace detail {
 
 //------------------------------------------------------------------------------
 
+template<typename DynamicBuffer>
+boost::optional<typename DynamicBuffer::mutable_buffers_type>
+prepare_or_error(DynamicBuffer& buffer, error_code& ec)
+{
+    try
+    {
+        ec.assign(0, ec.category());
+        return {buffer.prepare(read_size_or_throw(buffer, 65536))};
+    }
+    catch(const std::length_error& error)
+    {
+        ec = error::buffer_overflow;
+        return {};
```

Comment:
Does this return `boost::none` or does it return an engaged, default-constructed `mutable_buffers_type`?